### DISCUSSION TITLE
Fixing a subdomain to be a specific value

### DISCUSSION THREAD

**Discussion by schwarzschlyle** [lululxvi/deepxde#1448] [Q&A]
Hello. I would like to ask for help. I want to fix a 1x1 cube to have a value of one and try to find the function that minimizes the loss on the 10x10 cube which encloses the 1x1 cube. At the edge of the 10x10 cube, I want it to have a value of 0. Am I doing this correctly?

    space = dde.geometry.geometry_3d.Cuboid([-10,-10,-10],[10,10,10])
    side = 1
    passenger = dde.geometry.geometry_3d.Cuboid([-side,-side,-side],[side,side,side])
    geom = space - passenger

    
    def boundary_inner(x, inside):
        return inside and passenger.on_boundary(x)
    
    
    def boundary_outer(x, on_boundary):
        return on_boundary and geom.on_boundary(x)
    

    asymptotic_bc = dde.icbc.DirichletBC(geom, 
                                        lambda x: 0, 
                                        boundary_outer)
    
    
    passenger_bc = dde.icbc.DirichletBC(passenger, 
                                        lambda x: 1, 
                                        boundary_inner)
    
    

    
    data = dde.data.PDE(
        geom, pde, [asymptotic_bc,
                   passenger_bc], 
                   num_domain=domain,
                   num_boundary=boundary)

    net = dde.nn.FNN([3] + [width] * depth + [1], "tanh", "Glorot normal")

    model = dde.Model(data, net)
    model.compile("adam", lr=rate)
    model.train(iterations = 10000)
    model.compile("L-BFGS")
    losshistory, train_state = model.train(iterations=epochs)
    dde.saveplot(losshistory, train_state, issave=False, isplot=True)

  
    
    print("Training Complete!")

**Comment by lululxvi** [comment]
Seems correct.
